vignettes/creating-a-model.Rmd

```diff
@@ -0,0 +1,148 @@
+  purrr::map_chr(~ stringr::str_remove(., stringr::fixed(".Diseasy"))) |>
+  purrr::map_chr(~
+    paste0(
+      "- `$", stringr::str_to_lower(.), "`",
+      ": `{Diseasy", ., "}`"
+    )
+  ) |>
+  cat(sep = "\n")
+```
+
+This means, that once a new `DiseasyModel` instance is created, it will have the functional modules internally
+available. Going back to the above example of the SIR model with seasonal forcing, the model template would be
+access the seasonal forcing `s(t)` via the call `self$season$model_t(t)`.
+```{r check_DiseasySeason_function_signature, include = FALSE}
+# If the the signature of DiseasySeason is changed, please update this vignette
+stopifnot("model_t" %in% names(DiseasySeason$active))
+```
+
+## Training data
+The `DiseasyModel` class also contains the method `$get_training_data()` that provides training data to the models.
+
+This method interfaces with the `DiseasyObservables` module to provide data for an observable at some stratification
+level. The level of stratification is generally a flexible parameter that will be set by the user when requesting
+results from the model. For example, the user my request results at the country level or at the regional level.
+
+The helper method `$get_training_data()` will then provide the training data for the observable at the requested level.
+This data will contain one record for each time point in the training period, with columns for the date of the
+observation, the values of the different (optional) stratification, and the value of the observable on the date, and
+finally, an integer counter for the day relative to the end of the training period.
+
+## Model-user interface
+Beyond providing access to the functional modules, `DiseasyModel` defines how the user will interface with the model.
+
+### Setting parameters
+The parameters of the model should be contained within the `$parameters` slot of the model template.
+
+These are then exposed to the user, and logic can be implemented to allow the user to set some parameters of the model
+while leaving others free to be determined by the model during model fitting.
+
+### Getting results
+Once the parameters of the model is set (either by the user or by the model), the user can query the model for results
+using the `$get_results()` function. Through this function, the user provides the parameters of the result that should
+be provided (target observable, time-window and stratification level). In return, the function returns a standardised
+`data.frame`-like output that can be used for further analysis in conjunction with the results from other models.
+
+The outputs of the model are standardised, and should follow the structure of the `training_data` provided by
+`$get_training_data()` enriched by model prediction, model id and realization id.
+
+[^1]: The seasonal model exposed by a configured `DiseasySeason` instance is "closure" instead of a pure function.
```

Comment:
-> 'a closure'
```suggestion
[^1]: The seasonal model exposed by a configured `DiseasySeason` instance is a "closure" instead of a pure function.
```